apis/mutations/v1alpha1/assignmetadata_types.go

```diff
@@ -0,0 +1,89 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// AssignMetadataSpec defines the desired state of AssignMetadata
+type AssignMetadataSpec struct {
+	Match      Match      `json:"match,omitempty"`
+	Location   string     `json:"location,omitempty"`
+	Parameters Parameters `json:"parameters,omitempty"`
+}
+
+// AssignMetadataStatus defines the observed state of AssignMetadata
+type AssignMetadataStatus struct {
+	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+}
+
+type Match struct {
+	Kinds              []Kinds                            `json:"kinds,omitempty"`
```

Comment:
Should this be a simple item and not a slice?

Reply:
Checked on a generated CRD, it's' an array

Reply:
I'll take it as - leave as it is